

geometry_msgs/Pose2D[] state_ref  # state_reference

mpc_track/control[] control_ref  # control_reference

geometry_msgs/Pose2D state  #car_state

# request
---
# respond

mpc_track/control opt_control  #control valve 

geometry_msgs/Pose2D[] state_pre #state_predictive

float32 solve_time #optimize time
